### DISCUSSION TITLE
Help using zz after scroll with mini.animate

### DISCUSSION THREAD

**Discussion by diego-velez** [nvim-mini/mini.nvim#1436] [Q&A] [answered]
Hello!

I'm trying out `mini.animate`, and I'm having some trouble with my keymaps. In particular, my `<C-d>zz` and `<C-u>zz` keymaps. I'm aware that I should be using `execute_after` but I can't get it to work.

This is what I currently have:
```
vim.keymap.set(
  'n',
  '<C-d>',
  '<Cmd>lua vim.cmd("exe "normal! \\<C-d>""); '
    .. 'MiniAnimate.execute_after("scroll", "normal! zz")<CR>'
)
vim.keymap.set(
  'n',
  'n',
  '<Cmd>lua vim.cmd("normal! n"); ' .. 'MiniAnimate.execute_after("scroll", "normal! zzzv")<CR>'
)
vim.keymap.set(
  'n',
  'N',
  '<Cmd>lua vim.cmd("normal! N"); ' .. 'MiniAnimate.execute_after("scroll", "normal! zzzv")<CR>'
)
vim.keymap.set(
  'n',
  'G',
  '<Cmd>lua vim.cmd("normal! G"); ' .. 'MiniAnimate.execute_after("scroll", "normal! zz")<CR>'
)
```

The first keymap is not working, but the others are.

**Comment by diego-velez** [comment]
Also, it seems to lag when I reach the end of a file while holding `<C-d>`. I did it with default `<C-d>` mapping. Same is true for `<C-u>` and reaching the top of a file.

**Comment by echasnovski** [comment, marked as answer]
> The first keymap is not working, but the others are.

This is because it executes `vim.cmd("exe "normal! \\<C-d>"")` which both not syntactically correct and (probably) won't work with `\\<C-d>` as the argument to `normal!`. Use `'<Cmd>lua vim.cmd("normal! \4"); '` (`\4` is the "true" termcode/keycode for `<C-d>`) as the first part of the mapping's RHS.

But I'd say that any centering after scroll will have limited support, unfortunately. Mostly due to the fact that scroll is done with side effects (basically, what is said [here](https://github.com/echasnovski/mini.nvim/blob/2faada1cffce5609d78ac5e81eaf3754e0476758/doc/mini-animate.txt#L279-L295)) and the "general" nature of how scrolling is implemented.
